tests/safety/test_subaru.py

```diff
@@ -101,6 +104,52 @@ class TestSubaruGen2SafetyBase(TestSubaruSafetyBase):
   MAX_RATE_DOWN = 40
   MAX_TORQUE = 1000
 
+
+class TestSubaruLongitudinalSafetyBase(TestSubaruSafetyBase):
+  MIN_GAS = 808
+  MAX_GAS = 3400
+  INACTIVE_GAS = 1818
+  MAX_POSSIBLE_GAS = 2**12
+
+  MIN_BRAKE = 0
+  MAX_BRAKE = 600
+  MAX_POSSIBLE_BRAKE = 2**16
+
+  FWD_BLACKLISTED_ADDRS = {2: [MSG_SUBARU_ES_LKAS, MSG_SUBARU_ES_Brake, MSG_SUBARU_ES_Distance,
+                               MSG_SUBARU_ES_Status, MSG_SUBARU_ES_DashStatus,
+                               MSG_SUBARU_ES_LKAS_State, MSG_SUBARU_ES_Infotainment]}
+
+  def _generic_limit_safety_check(self, msg_function, min_value, max_value, max_possible_value, inactive_value=0):
```

Comment:
doesn't seem subaru specific. @sshane do we have something like this somewhere else?

Reply:
I started one here: https://github.com/commaai/panda/pull/1537

RPM would be specific to subaru, and some cars use an acceleration request for brake, and throttle-like message for gas. But I can look at cleaning it up into a common test

Reply:
yeah let's do that first

Reply:
done https://github.com/commaai/panda/pull/1537